### PR TITLE
Fix some typos and issues in haddock and user guide

### PR DESCRIPTION
**Pull request by jhrcek**
Thanks for your continued work on this great project. I enjoy using it very much!
Here's just a few typo and docs fixes I stumbled upon when re-reading the user guide.

### CODE REVIEW

File: docs/guide.rst [hunk 1]
```diff
@@ -941,12 +941,12 @@ in event handlers in ``EventM``:
      mExtent <- Brick.Main.lookupExtent FooBox
      case mExtent of
        Nothing -> ...
-       Just (Extent _ upperLeft (width, height) offset) -> ...
+       Just (Extent _ upperLeft (width, height)) -> ...
```
Comment (jhrcek) [inline docs/guide.rst:944]: This field has been removed in https://github.com/jtdaugherty/brick/commit/249781a57b3c28ea50529a7f8d0eb3c8470dd433

File: src/Brick/Widgets/Core.hs [hunk 2]
```diff
@@ -843,9 +843,8 @@ withAttr an p =
       withReaderT (ctxAttrNameL .~ an) (render p)
 
 -- | Update the attribute map while rendering the specified widget: set
--- its new default attribute to the one that we get by looking up the
```
Comment (jhrcek) [inline src/Brick/Widgets/Core.hs:846]: The original comment referred to some attribute name, but this function is not taking any attribute name as input..

File: src/Brick/Widgets/ProgressBar.hs [hunk 3]
```diff
@@ -10,7 +10,6 @@ module Brick.Widgets.ProgressBar
 where
 
 import Lens.Micro ((^.))
-import Data.List (splitAt)
```
Comment (jhrcek) [inline src/Brick/Widgets/ProgressBar.hs:13]: Unrelated to language fixes - just fixing an unused import warning (splitAt is exported from Prelude)

**Comment by jtdaugherty** [comment]
Thank you!

**Pull request merged**